diff --git a/CMakeLists.txt b/CMakeLists.txt
index 998db15..b942d90 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,22 @@ if(NOT CMake_TEST_EXTERNAL_CMAKE)
       "Use CMake 3.9 or lower instead."
       )
   endif()
+  if(CMAKE_SYSTEM_NAME STREQUAL "QNX")
+    list(APPEND uv_libraries
+      )
+    list(APPEND uv_headers
+      include/uv-posix.h
+      )
+    list(APPEND uv_defines
+      )
+    list(APPEND uv_sources
+      src/unix/bsd-ifaddrs.c
+      src/unix/no-fsevents.c
+      src/unix/no-proctitle.c
+      src/unix/posix-hrtime.c
+      src/unix/posix-poll.c
+      )
+  endif()
 
   set(CMake_BIN_DIR ${CMake_BINARY_DIR}/bin)
 endif()
